packages/@aws-cdk/aws-appsync/lib/graphqlapi.ts

```diff
@@ -518,23 +519,20 @@ export class GraphQLApi extends Construct {
     };
   }
 
-  private createAPIKey(config: ApiKeyConfig) {
-    let expires: number | undefined;
-    if (config.expires) {
-      expires = new Date(config.expires).valueOf();
-      const days = (d: number) =>
-        Date.now() + Duration.days(d).toMilliseconds();
-      if (expires < days(1) || expires > days(365)) {
-        throw Error('API key expiration must be between 1 and 365 days.');
-      }
-      expires = Math.round(expires / 1000);
+  private createAPIKey(config: ApiKeyConfig): string {
+    if (config.expires &&
+      (config.expires?.date < Expires.after(Duration.days(1)).date
```

Comment:
Feels like it could be a useful method on `Expires`:

```ts
if (config.expires?.lessThan(Duration.days(1)) || config?.expires.greaterThan(Duration.days(365)) {
  throw new Error(...)
}